node/request.js

```diff
@@ -188,6 +190,29 @@ TChannelRequest.prototype.resend = function resend() {
     }
 };
 
+TChannelRequest.prototype.checkTimeout = function checkTimeout(err, res) {
+    var self = this;
+    var now = self.timers.now();
+    self.elapsed = now - self.start;
+    if (self.elapsed < self.timeout) return false;
+    if (err) {
+        if (!self.err) {
+            self.emit('error', err);
+        }
+    } else if (res) {
+        if (!self.err && !self.res) {
+            self.emit('response', res);
+        }
+    } else if (!self.err) {
+        self.emit('error', errors.TimeoutError({
```

Comment:
it's not dead, it's used by the two new call sites circa L134 and L148 as added by https://github.com/uber/tchannel/commit/57e60e81f64e1866fb4737af9923578288284aa1